### ISSUE TITLE
Add validation error for broken internal links in components

### ISSUE THREAD

**Issue by duffy-james** [CityOfBoston/boston.gov-d8#1357]
There's been a consistent issue on the site where content authors attempt to add internal links this way on the site:
![image](https://user-images.githubusercontent.com/7929556/81824676-befe5180-9503-11ea-854a-b1da34a4ef7b.png)

Looking at the structure, it would appear to work. You're adding the relative path to the link. However, it doesn't work, plus, a user is able to save the page as a draft. This results in a broken page:

![image](https://user-images.githubusercontent.com/7929556/81824993-113f7280-9504-11ea-8684-dc424cbde189.png)

There's no indication in Drupal what is wrong for the user, and they never get a message before saving that what they're doing is valid. I'm also forced to deconstruct the page to see what's wrong, which is incredibly frustrating.

To address this issue, could we please either:

1. add a validation error that prevents a user from saving in this situation, forcing them to search for the link, or
2. allow a user to add this relative path (as we see in the example above) and make the link valid.

**Comment by reillyzlab** [comment]
Related issue: https://github.com/CityOfBoston/boston.gov-d8/issues/691

**Comment by padora-cob** [comment]
Should be fixed with a adjustment to some of our custom code, now using the Core Link fields stored URI route to render the links.

**Comment by rachelsater** [comment]
@duffy-james can you please check this is ok?

**Comment by duffy-james** [comment]
@padora-cob I tried adding random text and saw the error message. When I added a relative path and saved, it also looked like it save properly and worked. The only thing I noticed when testing is that, when I tried to add the relative path to a "transactions grid" component, it saved the page but the link is still broken.

[Here's the I'm testing](https://d8-ci.boston.gov/node/7536/latest), and the broken link is here:

![image](https://user-images.githubusercontent.com/7929556/130262392-3236da5b-63f3-440d-b657-130db55bd9e6.png)

Should I run through and test each component?
